packages/babel-parser/src/parser/expression.js

```diff
@@ -829,13 +829,13 @@ export default class ExpressionParser extends LValParser {
   ): N.Expression {
     if (node.callee.type === "Import") {
       if (node.arguments.length === 2) {
-        this.expectPlugin("moduleAttributes");
+        this.expectPlugin("importAssertions");
```

Comment:
We can't just rename the parser plugin since that would be a breaking change. They need to live side-by-side until Babel 8.

Reply:
I think it is okay because it will be rather confusing if error is thrown as
```
You will need one of `moduleAttributes` and `importAssertions` to parse this experimental syntax
```
And then users just find out `syntax-module-attributes` has been deprecated.

Reply:
We can tweak the error message, but currently this code works:
```js
parser.parse("import 'foo' with type: 'json'", { plugins: ["moduleAttributes"] });
```
it cannot just stop working in a minor release.

Reply:
we will need to keep the support for the old `with` moduleAttributes and the new `assert` importAssertions side-by-side in the parser. I'm happy to break people that rely on moduleAttributes to reduce code complexity, but if we need to we can support both

Reply:
I'd like to remove the old code too, but we cannot do it in a minor release (we cannot _knowingly_ introduce a breaking change).